keel-web/src/test/kotlin/com/netflix/spinnaker/keel/rest/DeliveryConfigYamlParsingFilterTests.kt

```diff
@@ -17,26 +23,67 @@ import io.mockk.just
 import io.mockk.mockk
 import io.mockk.slot
 import io.mockk.verify
-import javax.servlet.FilterChain
-import javax.servlet.ServletRequest
-import javax.servlet.http.HttpServletRequestWrapper
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.jackson.JsonComponentModule
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment.NONE
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.ComponentScan
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Primary
 import org.springframework.mock.web.MockHttpServletRequest
 import org.springframework.mock.web.MockHttpServletResponse
 import strikt.api.expectThat
 import strikt.assertions.hasSize
 import strikt.assertions.isA
 import strikt.assertions.isEqualTo
+import javax.servlet.FilterChain
+import javax.servlet.ServletRequest
+import javax.servlet.http.HttpServletRequestWrapper
 
+@SpringBootTest(
+  classes = [Ec2JsonTestConfiguration::class],
+  webEnvironment = NONE
+)
 class DeliveryConfigYamlParsingFilterTests : JUnit5Minutests {
```

Comment:
@robfletcher This test is currently failing, even with the configuration below., with this message:
```
com.fasterxml.jackson.databind.JsonMappingException: Cannot construct instance of `com.netflix.spinnaker.keel.core.api.SubmittedResource`, problem: Cannot construct instance of `com.netflix.spinnaker.keel.api.ec2.SecurityGroupRule` (no Creators, like default constructor, exist): abstract types either need to be mapped to concrete types, have custom deserializer, or contain additional type information
```
I've tried adding a `testRuntimeOnly` dependency to `keel-ec2-plugin`, since the package scanned for deserializers is in there, but no luck. Any ideas?

Reply:
Issue was the usage of `@ConditionalOnMissingBean` on `SecurityGroupRuleDeserializer`. It doesn't work like I thought it did, and the bean is never instantiated even when there's no other one of the same type.